src/plugins/kibana_react/public/app_links/utils.test.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getClosestLink } from './utils';
+
+const createBranch = (...tags: string[]): HTMLElement[] => {
```

Comment:
It might be simpler to just use raw html strings. I prefer this because it eliminates the possibility of there being a bug in this setup code.

```ts
const element = document.createElement('div')
element.innerHTML = `<div><a></a></div>`;
```

Reply:
This utility method has the advantage to return the created branch elements as an array
```ts
const [, container, target] = createBranch('A', 'DIV', 'SPAN');
```
Using plain `innerHTML` would force to manually access each individual node of the branch in the tests. I think the decreased readability is a biggest downside than a bug risk in the `createBranch` method?

I could also change to something like
```ts
const element = document.createElement('a')
element.innerHTML = `<div><span></span></div>`;
const [, container, target] = getLeftmostBranch(element);
```
but it's still 3 lines versus one (and `getLeftmostBranch` would have a logic very close to `createBranch` anyway)

wdyt?